sentry_sdk/integrations/chalice.py

```diff
@@ -0,0 +1,157 @@
+
+import chalice
+from chalice import Chalice, ChaliceViewError, Response
+from chalice.app import EventSourceHandler as ChaliceEventSourceHandler
+from sentry_sdk._types import MYPY
+from sentry_sdk.hub import Hub, _should_send_default_pii
+from sentry_sdk.integrations import Integration
+from sentry_sdk.integrations._wsgi_common import _filter_headers
+from sentry_sdk.integrations.aws_lambda import _get_cloudwatch_logs_url
+from sentry_sdk.utils import (
+    AnnotatedValue,
+    capture_internal_exceptions,
+    event_from_exception,
+)
+
+if MYPY:
+    from typing import Any, Optional
+
+    from sentry_sdk._types import Event, EventProcessor, Hint
+
+
+class EventSourceHandler(ChaliceEventSourceHandler):
+    def __call__(self, event, context):
+        hub = Hub.current
+        client = hub.client
+
+        with hub.push_scope() as scope:
+            try:
+                event_obj = self.event_class(event, context)
+                return self.func(event_obj)
+            except Exception:
+                scope.add_event_processor(_make_request_event_processor(event, context))
+                exc_info = sys.exc_info()
+                event, hint = event_from_exception(
+                    exc_info,
+                    client_options=client.options,
+                    mechanism={"type": "chalice", "handled": False},
+                )
+                hub.capture_event(event, hint=hint)
+                hub.flush()
+
+
+def _get_view_function_response(app, view_function, function_args):
+    hub = Hub.current
+    client = hub.client
+
+    with hub.push_scope() as scope:
+        try:
+            response = view_function(**function_args)
+            if not isinstance(response, Response):
+                response = Response(body=response)
+            app._validate_response(response)
+        except ChaliceViewError as e:
+            # Any chalice view error should propagate.  These
+            # get mapped to various HTTP status codes in API Gateway.
+            response = Response(
+                body={"Code": e.__class__.__name__, "Message": str(e)},
+                status_code=e.STATUS_CODE,
+            )
+            hub.flush()
+        except Exception:
+            with capture_internal_exceptions():
+                scope.transaction = app.lambda_context.function_name
+                scope.add_event_processor(
+                    _make_request_event_processor(
+                        app.current_request, app.lambda_context
+                    )
+                )
+            exc_info = sys.exc_info()
+            event, hint = event_from_exception(
+                exc_info,
+                client_options=client.options,
+                mechanism={"type": "chalice", "handled": False},
+            )
+            hub.capture_event(event, hint=hint)
+            hub.flush()
+            headers = {}
+            app.log.error("Caught exception for %s", view_function, exc_info=True)
```

Comment:
I really don't think the SDK is supposed to do that. Just re-raise the exception.

Reply:
i made a monkeypatch of the '_get_view_function_response' function from the chalice, in order to keep the uniformity of the code (a good practice of chalice), I keep this line https://github.com/aws/chalice/blob/2e44fe3b63378b488af88bd92254c58708cb9ac2/chalice/app.py#L1123

Reply:
Oh that's not great, copypasting the code. What about wrapping view_function, and passing the wrapped function into the original implementation of `_get_view_function_response`?

Reply:
Why don't you have access to the app context?

```python
def _get_view_function_response(app, view_function, ...):
    def new_view_function(**kw):
        # you can access app.lambda_context here
        try:
            return view_function(**kw)
        except Exception:
            ...

    return old_get_view_function_response(app, new_view_function, ...)

Reply:
I solved the last problem about the code copied from chalice and this is my suggestion https://github.com/getsentry/sentry-python/blob/59f092d760da9ef888616e3c9cb7a53ece382f73/sentry_sdk/integrations/chalice.py#L48 and https://github.com/getsentry/sentry-python/blob/59f092d760da9ef888616e3c9cb7a53ece382f73/sentry_sdk/integrations/chalice.py#L69, could be another way to wrap the original function _get_view_function_response.